### PR TITLE
AT: Update eligibility holds to match the ones coming from API

### PR DESCRIPTION
**Pull request by vindl**
Resolves: https://github.com/Automattic/wp-calypso/issues/11889

Updates the eligibility blocker constants and messages to match the current ones that are returned from  WordPress.com API.

### CODE REVIEW

File: client/blocks/eligibility-warnings/index.jsx [hunk 1]
```diff
@@ -57,14 +72,19 @@ function getHoldMessages( translate ) {
...
+			title: translate( 'No WordPress.com name servers' ),
+			description: translate( 'Your custom domain must point to WordPress.com name servers.' ),
 			supportUrl: 'https://support.wordpress.com/'
 		},
-		SITE_PRIVATE: {
-			title: translate( 'Private site' ),
-			description: translate( 'This feature is not supported on private sites.' ),
+		NOT_RESOLVING_TO_WPCOM: {
+			title: translate( 'Primary domain not pointing to WordPress.com servers' ),
+			description: translate( 'Your primary domain must point to WordPress.com name servers.' ),
```
Comment (a8ci18n) [inline client/blocks/eligibility-warnings/index.jsx]: Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'Your custom domain must point to WordPress.com name servers.' )` *ES Score: 12*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/eligibility-holds-messages)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/blocks/eligibility-warnings/index.jsx [hunk 2]
```diff
@@ -27,19 +27,34 @@ function getHoldMessages( translate ) {
...
+			supportUrl: 'https://support.wordpress.com/'
+		},
 		NO_BUSINESS_PLAN: {
 			title: translate( 'Business plan required' ),
 			description: translate( 'This feature is only allowed on sites with a business plan.' ),
 			supportUrl: 'https://support.wordpress.com/'
 		},
+		NO_JETPACK_SITES: {
+			title: translate( 'Jetpack site' ),
+			description: translate( 'This feature is not supported on Jetpack sites.' ),
```
Comment (a8ci18n) [inline client/blocks/eligibility-warnings/index.jsx]: Hi! I've found a possible matching string that has already been translated **2** times:
`translate( 'This feature is not supported on VIP sites.' )` *ES Score: 9*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/eligibility-holds-messages)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/blocks/eligibility-warnings/index.jsx [hunk 3]
```diff
@@ -27,19 +27,34 @@ function getHoldMessages( translate ) {
...
+			description: translate( 'Another installation is already in progress.' ),
+			supportUrl: 'https://support.wordpress.com/'
+		},
 		NO_BUSINESS_PLAN: {
 			title: translate( 'Business plan required' ),
 			description: translate( 'This feature is only allowed on sites with a business plan.' ),
 			supportUrl: 'https://support.wordpress.com/'
 		},
+		NO_JETPACK_SITES: {
+			title: translate( 'Jetpack site' ),
```
Comment (a8ci18n) [inline client/blocks/eligibility-warnings/index.jsx]: Hi! I've found a possible matching string that has already been translated **16** times:
`translate( 'Manage Jetpack Sites' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/blocks/eligibility-warnings/index.jsx [hunk 4]
```diff
@@ -27,19 +27,34 @@ function getHoldMessages( translate ) {
 			description: '',
 			supportUrl: '',
 		},
+		TRANSFER_ALREADY_EXISTS: {
+			title: translate( 'Installation in progress' ),
```
Comment (a8ci18n) [inline client/blocks/eligibility-warnings/index.jsx]: Hi! I've found a possible matching string that has already been translated **31** times:
`translate( 'Installation', { context: 'Navigation item'} )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/blocks/eligibility-warnings/index.jsx [hunk 5]
```diff
@@ -27,45 +27,65 @@ function getHoldMessages( translate ) {
...
+			description: translate( 'Your custom domain must point to WordPress.com name servers.' ),
+			supportUrl: 'https://support.wordpress.com/domain-helper/'
 		},
-		SITE_PRIVATE: {
-			title: translate( 'Private site' ),
-			description: translate( 'This feature is not supported on private sites.' ),
-			supportUrl: 'https://support.wordpress.com/'
+		NOT_RESOLVING_TO_WPCOM: {
+			title: translate( 'Primary domain not pointing to WordPress.com servers' ),
+			description: translate( 'Your primary domain must point to WordPress.com servers.' ),
```
Comment (a8ci18n) [inline client/blocks/eligibility-warnings/index.jsx]: Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'Your custom domain must point to WordPress.com name servers.' )` *ES Score: 11*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by lamosty** [comment]
I forgot to add: I think the messaging could be improved a little bit. Currently, we say:

"Error: Primary domain does not have a valid SSL certificate" -- this is okay.
"Your primary domain must have a valid SSL certificate to use this feature." -- this is not as users don't know what to do. "What is SSL? Why am I seeing this? I'll contact support".

We need to tell them to wait as they will get the ssl after some time. /cc @rralian for wording.

**Comment by roundhill** [comment]
How long does it take for the SSL certificate to kick in? Maybe the sub-text could be:

`We're still setting up some security settings on the site. Try again in a few minutes.`

**Comment by lamosty** [comment]
> How long does it take for the SSL certificate to kick in? Maybe the sub-text could be:

When I tested it, it's at least 10 minutes, sometimes even a bit more. Don't know whether I would call that "a few minutes" but yeah, we should be optimistic. :)

**Comment by roundhill** [comment]
> When I tested it, it's at least 10 minutes, sometimes even a bit more. Don't know whether I would call that "a few minutes" but yeah, we should be optimistic. :)

We could just do the generic `Try again later.`

**Comment by lamosty** [comment]
> We could just do the generic Try again later.

Yeah.

My point is that we should clearly say to the customer that they don't need to do anything. I would even hide the Resolve button in this case as it's not their mistake. It's simply how wp.com sites work in general.

**Comment by lamosty** [comment]
Every WP.com business customer will see this screen with this "conflict" after creating a new site and purchasing a domain and directly trying to install a plugin or a custom theme after that.

**Comment by roundhill** [comment]
Oh, if they don't need to do anything maybe we could say something like:

`You will be able to proceed once we finish setting up some security settings for the site.`

**Comment by lamosty** [comment]
> You will be able to proceed once we finish setting up some security settings for the site.

Seems good enough for me. :)

I would also drop the "Error:" part and remove the "Resolve" button. If possible, even change the red "!" icon to something less scary (for this we may need a designer's eye).

**Comment by vindl** [comment]
> I would also drop the "Error:" part and remove the "Resolve" button. If possible, even change the red "!" icon to something less scary (for this we may need a designer's eye).

@lamosty there is already an in-progress PR that redesigns this component completely, and changes the things that you've mentioned, so no need to do this here.

**Comment by rralian** [comment]
> You will be able to proceed once we finish setting up some security settings for the site.

I'm not in love with that text because it implies my site isn't yet secure. That sounds bad. :-) But let's not haggle on wording here. Let's just get the functionality in place for now. Then I think we should probably do a general review of all of the error/warning wording all at once prior to launch. I'd want to get @apeatling's opinion on the messaging across all of the scenarios.

**Comment by Copons** [comment]
Just FYI, the proper copy is slated for a review after https://github.com/Automattic/wp-calypso/pull/11575 is merged.
In the new version some holds will be converted in banner nudges (for now: `NO_BUSINESS_PLAN` and `NOT_USING_CUSTOM_DOMAIN`), others will likely be reworded in a more friendly way.

**Pull request merged**